ingest/scripts/call_loculus.py

```diff
@@ -173,73 +172,110 @@ def get_or_create_group_and_return_group_id(config: Config, allow_creation: bool
     return create_group_and_return_group_id(config)
 
 
-def filter_fasta_by_submission_ids(
-    fasta_dict: dict[str, Any], submission_ids: list[str], config: Config
-):
-    if config.segmented:
-        segmented_submission_ids = [
-            f"{submission_id}_{segment}"
-            for submission_id in submission_ids
-            for segment in config.nucleotide_sequences
-        ]
-        return [
-            fasta_dict[submission_id]
-            for submission_id in segmented_submission_ids
-            if submission_id in fasta_dict
-        ]
-    return [
-        fasta_dict[submission_id] for submission_id in submission_ids if submission_id in fasta_dict
-    ]
-
-
 def post_fasta_batches(
     url,
     fasta_file: str,
     metadata_file: str,
     config: Config,
     params: dict[str, str],
-    chunk_size=60000,
+    chunk_size=10000,
 ) -> requests.Response:
     """Chunks metadata files, joins with sequences and submits each chunk via POST."""
-    df = pd.read_csv(metadata_file, sep="\t")
-    logger.info(df.columns)
-    sequences_dict = {record.id: record for record in SeqIO.parse(fasta_file, "fasta")}
-    submission_ids = df["submissionId"].tolist()
-    for batch_num, submission_id_chunk in enumerate(
-        submission_ids[i : i + chunk_size] for i in range(0, len(submission_ids), chunk_size)
+
+    def submit(metadata_batch_output, sequences_batch_output, number_of_submissions):
+        batch_num = -(number_of_submissions // -chunk_size)  # ceiling division
+        logger.info(f"Submitting batch {batch_num}")
+
+        metadata_in_memory = BytesIO("".join(metadata_batch_output).encode("utf-8"))
+        fasta_in_memory = BytesIO("".join(sequences_batch_output).encode("utf-8"))
+
+        files = {
+            "metadataFile": ("metadata.tsv", metadata_in_memory, "text/tab-separated-values"),
+            "sequenceFile": ("sequences.fasta", fasta_in_memory, "text/plain"),
+        }
+        response = make_request(HTTPMethod.POST, url, config, params=params, files=files)
+        logger.info(f"Batch {batch_num} Response: {response.status_code}")
+        if response.status_code != 200:
+            logger.error(f"Error in batch {batch_num}: {response.text}")
+
+        return response
+
+    number_of_submissions = -1
+    fasta_submission_id = None
+    fasta_record_header = None
+    metadata_header = None
+
+    sequences_batch_output = []
+    metadata_batch_output = []
+
+    with (
+        open(fasta_file, encoding="utf-8") as fasta_file_stream,
```

Comment:
potentially replace with seqIOs SeqIO.parse() which should read in the file record by record and handle multi-line sequences which will not be correctly handled in this case